stream_read.py

```diff
@@ -0,0 +1,484 @@
+"""Module for parsing indexing stream file produced by CrystFEL indexamajig.
+"""
+import numpy as np
+import sys
+
+
+def cell_parameters(astar, bstar, cstar):
+    """Calculates unit cell parameters.
+
+    Parameters
+    ----------
+    astar : list
+
+        List of float values.
+        (A line beginning with `astar`)
+    bstar : list
+
+        List of float values.
+        (A line beginning with `bstar`)
+    cstar : list
+
+        List of float values.
+        (A line beginning with `cstar`)
+
+    Returns
+    -------
+    a, b, c, alfa, beta, gamma : tuple
+
+        Unit cell parameters.
+    """
+
+    def angle_between(x1, x2):
+        """Auxiliary function for calculating the angle between vectors.
+
+        Parameters
+        ----------
+        x1 : numpy.ndarray
+
+            First vector.
+        x2 : numpy.ndarray
+
+            Second vector.
+
+        Returns
+        -------
+        Angle in degrees.
+        """
+        # magnitude of a vector x1
+        mod1 = np.linalg.norm(x1)
+        # magnitude of a vector x2
+        mod2 = np.linalg.norm(x2)
+        cosine = (np.sum(x1*x2)) / (mod1*mod2)
+        if (cosine > 1.0):
+            cosine = 1.0
+        if (cosine < -1.0):
+            cosine = -1.0
```

Comment:
You can use `np.clip` here:

```
cosine = np.clip(cosine, -1, 1)
```

Reply:
Done in: 871aef440c18590209993a1c2f17f6f757734098